File: frontend/src/app/controller/service/admin/commun/UniteMesureAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {UniteMesureDto} from 'src/app/controller/model/commun/UniteMesure.model';
import {UniteMesureCriteria} from 'src/app/controller/criteria/commun/UniteMesureCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class UniteMesureAdminService extends AbstractService<UniteMesureDto, UniteMesureCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/uniteMesure/');
    }

    public constrcutDto(): UniteMesureDto {
        return new UniteMesureDto();
    }

    public constrcutCriteria(): UniteMesureCriteria {
        return new UniteMesureCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/commun/UniteMesureAdminService.java
package ma.zs.easystock.service.facade.admin.commun;

import java.util.List;
import ma.zs.easystock.bean.core.commun.UniteMesure;
import ma.zs.easystock.dao.criteria.core.commun.UniteMesureCriteria;
import ma.zs.easystock.zynerator.service.IService;

import ma.zs.easystock.ws.dto.commun.UniteMesureDto;
import org.springframework.http.HttpEntity;


public interface UniteMesureAdminService extends  IService<UniteMesure,UniteMesureCriteria>  {



    HttpEntity<byte[]> createPdf(UniteMesureDto dto) throws Exception;

}
